app/models/comment.rb

```diff
@@ -84,12 +85,24 @@ def no_grandparents
     errors.add(:parent, 'cannot have a parent of their own') if parent&.parent
   end
 
+  def ama_closed
+    return unless post
+    return unless post.ama
+    ama = post.ama
+    return if ama.author == user
+    now_time = Time.now
+
+    unless ama.start_date <= now_time && ama.end_date >= now_time
+      errors.add(:post, 'cannot make any more comments on this ama')
+    end
+  end
+
```

Comment:
We still want to retain the functionality that posts/comments have for ama, and I'm not sure if we want a separate feeds for posts/comments that are under amas, but yeah it makes sense to condense this to something like `ama.closed?`